websockify/token_plugins.py

```diff
@@ -87,3 +87,49 @@ class JSONTokenApi(BaseTokenAPI):
     def process_result(self, resp):
         resp_json = resp.json()
         return (resp_json['host'], resp_json['port'])
+
+
+class JWTTokenApi(BasePlugin):
+    # source is a JWT-token, with hostname and port included
+    # Both JWS as JWE tokens are accepted. With regards to JWE tokens, the key is re-used for both validation and decryption.
+
+    def lookup(self, token):
+        try:
+            from jwcrypto import jwt
+            import json
+
+            key = jwt.JWK()
+            
+            try:
+                with open(self.source) as key_file:
```

Comment:
I would open the file in byte mode here. Nothing guarantees the key will only contain valid utf-8 characters. Unless you explicitely require the key to be hex-encoded, which is not the case here.

Reply:
Hmm, never thought of that. Will add 'rb' as mode :-) Thank you for your input!